zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/group/GroupTest.java

```diff
@@ -102,4 +103,81 @@ public void shouldRejectUpdatedIfSameGroupExists() {
             "Expected to update group with name '%s', but a group with this name already exists."
                 .formatted(updatedName));
   }
+
+  @Test
+  public void shouldAddEntityToGroup() {
+    final var userKey =
+        engine
+            .user()
+            .newUser("foo")
+            .withEmail("foo@bar")
+            .withName("Foo Bar")
+            .withPassword("zabraboof")
+            .create()
+            .getKey();
+    final var name = UUID.randomUUID().toString();
+    final var groupKey = engine.group().newGroup(name).create().getValue().getGroupKey();
+    final var updatedGroup =
+        engine
+            .group()
+            .addEntity(groupKey)
+            .withEntityKey(userKey)
+            .withEntityType(EntityType.USER)
+            .add()
+            .getValue();
+
+    Assertions.assertThat(updatedGroup)
+        .isNotNull()
+        .hasFieldOrPropertyWithValue("entityKey", userKey)
+        .hasFieldOrPropertyWithValue("entityType", EntityType.USER);
+  }
+
+  @Test
+  public void shouldRejectIfGroupIsNotPresentWhileAddingEntity() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var groupRecord = engine.group().newGroup(name).create();
+
+    // when
+    final var notPresentGroupKey = 1L;
+    final var notPresentUpdateRecord =
+        engine.group().addEntity(notPresentGroupKey).expectRejection().add();
+
+    final var createdGroup = groupRecord.getValue();
+    Assertions.assertThat(createdGroup).isNotNull().hasFieldOrPropertyWithValue("name", name);
+
+    assertThat(notPresentUpdateRecord)
+        .hasRejectionType(RejectionType.NOT_FOUND)
+        .hasRejectionReason(
+            "Expected to update group with key '"
+                + notPresentGroupKey
+                + "', but a group with this key does not exist.");
+  }
+
+  @Test
+  public void shouldRejectIfEntityIsNotPresent() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var groupRecord = engine.group().newGroup(name).create();
+
+    // when
+    final var createdGroup = groupRecord.getValue();
+    final var groupKey = createdGroup.getGroupKey();
+    final var notPresentUpdateRecord =
+        engine
+            .group()
+            .addEntity(groupKey)
+            .withEntityKey(1L)
+            .withEntityType(EntityType.USER)
+            .expectRejection()
+            .add();
+
+    Assertions.assertThat(createdGroup).isNotNull().hasFieldOrPropertyWithValue("name", name);
+
+    assertThat(notPresentUpdateRecord)
+        .hasRejectionType(RejectionType.NOT_FOUND)
+        .hasRejectionReason(
+            "Expected to add an entity with key '%s' and type '%s' to group with key '%s', but the entity doesn't exist."
```

Comment:
🔧 If you change the error message in the processor this should be aligned as well